app/services/mandataris.js

```diff
@@ -42,12 +44,26 @@ export default class MandatarisService extends Service {
     newMandatarisState,
     newFractie
   ) {
+    let mandatarisStatus = toReplace.status;
+
+    const mandaat = await toReplace.bekleedt;
     const existing = await this.getOverlappingMandate(
       toReplace,
       replacementPerson
     );
+
+    if (mandaat && mandaat.isBurgemeester) {
+      mandatarisStatus = this.store.findRecord(
+        'mandataris-status-code',
+        MANDATARIS_WAARNEMEND_STATE_ID
+      );
+    }
+
     if (existing) {
-      return existing;
+      const mandataris = await this.store.findRecord('mandataris', existing.id);
+      mandataris.status = mandatarisStatus;
```

Comment:
you can just use existing.status here, no?

Reply:
Hmm I'll try again but that didn't do it before

Reply:
It didn't update it if i do this
![image](https://github.com/user-attachments/assets/841f6ced-1e98-4ecf-816a-63a269f6cbb3)
```js
    if (existing) {
      // const mandataris = await this.store.findRecord('mandataris', existing.id);
      existing.status = mandatarisStatus;
      existing.save();
      console.log('existing', existing);
      console.log('existing', await existing.status);
      return existing;
    }

Reply:
hmmm that is weird
didn't run it yet no

Reply:
Figured it out, you can set it directly on the exisiting mandataris, but to do that, you need to await your findRecord on line 56. Not sure why this results in different behaviour with existing and the new fetch ... Maybe because you need to wait anyway if you fetch a new record?

Reply:
Jup works indeed ! Not sure why I didn't add the await before my store.findRecord :/